samples: Extract DTM api in sample (#11789)
nrfconnect/sdk-nrf: samples/bluetooth/direct_test_mode/README.rst

Reviewer: @dchat-nordic, could you please also fix a TYPO in line 300 in this file ("Anritsu MT885" -> "Anritsu MT8852")? Just spotted it when reading the docs and don't want to open a separate PR for that :)

The implementation is self-contained and requires no Bluetooth Low Energy protocol stack for its operation.
The MPU is initialized in the standard way.
The DTM library function :c:func:`dtm_init` configures all interrupts, timers, and the radio.

> The :file:`mai.c` file may be extended with other interface implementations, such as an HCI interface, USB, or another interface required by the Upper Tester.

Reviewer 2: ```suggestion
The :file:`main.c` file may be extended with other interface implementations, such as an HCI interface, USB, or another interface required by the Upper Tester.
```

DTM module interface
====================

The DTM module interface can be divided into three parts:

* Initialization function:

>   * ``dtm_init``

Reviewer 2: Good catch, that's correct.

* Setup functions:
  * ``dtm_setup_reset``
  * ``dtm_setup_set_phy``
  * ``dtm_setup_set_modulation``
  * ``dtm_setup_read_features``
  * ``dtm_setup_read_max_supported_value``
  * ``dtm_setup_set_cte_mode``
  * ``dtm_setup_set_cte_slot``
  * ``dtm_setup_set_antenna_params``
  * ``dtm_setup_set_transmit_power``
* Test functions:
  * ``dtm_test_receive``
  * ``dtm_test_transmit``

>   * ``dtm_test_end``

Reviewer 2: please add missing empty lines before and after sub-lists

DTM module interface
====================

> The DTM module interface can be divided into following three parts:

Reviewer 2: ```suggestion
The DTM module interface can be divided into the following three parts:
```

> * The :c:func:`dtm_init`` Initialization function

Reviewer 2: ```suggestion
* The :c:func:`dtm_init` initialization function
```

* Setup functions:

  * :c:func:`dtm_setup_reset`
  * :c:func:`dtm_setup_set_phy`
  * :c:func:`dtm_setup_set_modulation`
  * :c:func:`dtm_setup_read_features`
  * :c:func:`dtm_setup_read_max_supported_value`
  * :c:func:`dtm_setup_set_cte_mode`
  * :c:func:`dtm_setup_set_cte_slot`
  * :c:func:`dtm_setup_set_antenna_params`
  * :c:func:`dtm_setup_set_transmit_power`

* Test functions:

  * :c:func:`dtm_test_receive`
  * :c:func:`dtm_test_transmit`
  * :c:func:`dtm_test_end`

The setup functions implement the DTM setup required by the Bluetooth Low Energy standard.

> 

Reviewer 2: I recommend removing this empty line to have both similar sentences in one paragraph.

The DTM-to-Serial adaptation layer
==================================

> The :file:`main_uart.c` file is an implementation of the UART interface specified in the `Bluetooth Core Specification`_, Volume 6, Part F, Chapter 3.

Reviewer 3: This file no longer exists

The DTM-to-Serial adaptation layer
==================================

> The :file:`main_uart.c` file is an implementation of the UART interface specified in the `Bluetooth Core Specification`_: Vol. 6, Part F, Chap. 3.

Reviewer 2: ```suggestion
The :file:`main_uart.c` file is an implementation of the UART interface specified in the `Bluetooth Core Specification`_, Volume 6, Part F, Chapter 3.
```
